api/service.proto

```diff
@@ -0,0 +1,103 @@
+syntax = "proto3";
+
+import "github.com/kubernetes-csi/csi-proxy/api/file_system.proto";
+import "github.com/kubernetes-csi/csi-proxy/api/general_storage.proto";
+import "github.com/kubernetes-csi/csi-proxy/api/iscsi.proto";
+import "github.com/kubernetes-csi/csi-proxy/api/smb.proto";
+
+service CSIProxyService {
+    // ********************** //
+    // File system operations //
+    // ********************** //
+
+    // PathExists checks if the given path exists on the host.
+    rpc PathExists(PathExistsRequest) returns (PathExistsResponse) {}
+
+    // Mkdir creates a directory at the requested path on the host.
+    rpc Mkdir(MkdirRequest) returns (MkdirResponse) {}
+
+    // Rmdir removes the directory at the requested path on the host.
+    // This may be used for unlinking a symlink created through LinkVolume.
+    rpc Rmdir(RmdirRequest) returns (RmdirResponse) {}
+
+    // LinkPath invokes mklink on the global staging path of a volume linking it to a path within a container.
+    rpc LinkPath(LinkPathRequest) returns (LinkPathResponse) {}
+
+    // ********************************* //
+    // General storage/volume operations //
+    // ********************************* //
+
+    // Rescan refreshes the host storage cache.
+    rpc Rescan(RescanRequest) returns (RescanResponse) {}
+
+    // PartitionDisk initializes and partitions a disk device (if the disk has not
+    // been partitioned already) and returns the resulting volume object's GUID.
+    rpc PartitionDisk(PartitionDiskRequest) returns (PartitionDiskResponse) {}
+
+    // FormatVolume formats a volume with the provided file system.
+    // The resulting volume is mounted at the requested global staging path.
+    rpc FormatVolume(FormatVolumeRequest) returns (FormatVolumeResponse) {}
+
+    // ListDiskLocations returns locations <Adapter, Bus, Target, LUN ID> of all disk devices enumerated by Windows.
+    rpc ListDiskLocations(ListDiskLocationsRequest) returns (ListDiskLocationsResponse) {}
+
+    // ListDiskIDs returns all IDs (from IOCTL_STORAGE_QUERY_PROPERTY) of all disk devices enumerated by Windows.
+    rpc ListDiskIDs(ListDiskIDsRequest) returns (ListDiskIDsResponse) {}
+
+    // ListDiskVolumeMappings returns a map of all disk devices and volumes GUIDs
+    rpc ListDiskVolumeMappings(ListDiskVolumeMappingsRequest) returns (ListDiskVolumeMappingsResponse) {}
+
+    // ResizeVolume performs resizing of the partition and file system for a block based volume
+    rpc ResizeVolume(ResizeVolumeRequest) returns (ResizeVolumeResponse) {}
+
+    // DismountVolume gracefully dismounts a volume
+    rpc DismountVolume(DismountVolumeRequest) returns (DismountVolumeResponse) {}
```

Comment:
Is there a MountVolume call?

Reply:
Good catch. We need something along the lines of a `MountBlockVolume` (just like `MountSMBShare`). `MountBlockVolumeRequest` will need to comprise of:
```
string volume_device_guid 
string host_path
bool readonly
```
`MountBlockVolumeResponse` will comprise of `int32 error_code`.

Reply:
👍 adding